cloudify_azure/resources/app_service/webapp.py

```diff
@@ -0,0 +1,84 @@
+# #######
+# Copyright (c) 2016 GigaSpaces Technologies Ltd. All rights reserved
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from azure.common.credentials import ServicePrincipalCredentials
+from azure.mgmt.web import WebSiteManagementClient
+from cloudify.decorators import operation
+
+
+class WebApp(object):
+    def __init__(
+            self, logger, credentials, group_name, app_name, app_config=None):
+        self.group_name = group_name
+        self.app_name = app_name
+        self.logger = logger
+        self.app_config = app_config
+        self.resource_verify = bool(credentials.get('endpoint_verify', True))
+
+        self.credentials = ServicePrincipalCredentials(
+            client_id=str(credentials['client_id']),
+            secret=str(credentials['client_secret']),
+            tenant=str(credentials['tenant_id']),
+            verify=self.resource_verify
+        )
+        self.client = WebSiteManagementClient(
+            self.credentials, str(credentials['subscription_id']))
+        self.logger.info("Use subscription: {}"
+                         .format(credentials['subscription_id']))
+
+    def create_or_update(self):
+        """Deploy the template to a resource group."""
+        self.logger.info("Create/Updating web app...")
+        web_app_async = self.client.web_apps.create_or_update(
+            self.group_name,
+            self.app_name,
+            self.app_config
+        )
+        return web_app_async.result()
+
+    def get(self):
+        return self.client.web_apps.get(
+            resource_group_name=self.group_name, name=self.app_name)
+
+    def delete(self):
+        """Destroy the given resource group"""
+        self.logger.info("Delete web app...")
+        self.client.web_apps.delete(
+            resource_group_name=self.group_name,
+            name=self.app_name)
+
+
+@operation
+def create(ctx, resource_group, name, app_config, **kwargs):
+    if ctx.node.properties.get('use_external_resource', False):
+        ctx.logger.info("Using external resource")
```

Comment:
We should check that the web app exists, like run a get operation. If it fails or returns None, we should fail.